src/web/components/provider/LanguageProvider.tsx

```diff
@@ -0,0 +1,82 @@
+/* SPDX-FileCopyrightText: 2024 Greenbone AG
+ *
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import {
+  getLocale,
+  setLocale as changeGmpLocale,
+  onLanguageChange,
+} from 'gmp/locale/lang';
+import {createContext, useState, useEffect, useCallback, useMemo} from 'react';
+import useGmp from 'web/hooks/useGmp';
+
+interface LanguageContextProps {
+  language: string;
+  setLanguage: (lang: string) => Promise<void>;
+}
+
+interface LanguageProviderProps {
+  children: React.ReactNode;
+}
+
+const SETTING_ID_LOCALE = '6765549a-934e-11e3-b358-406186ea4fc5';
+
+export const LanguageContext = createContext<LanguageContextProps>({
+  language: '',
+  setLanguage: async () => {},
+});
+
+export const LanguageProvider: React.FC<LanguageProviderProps> = ({
+  children,
+}) => {
+  const gmp = useGmp();
+  const [languageState, setLanguageState] = useState<string>(
+    getLocale() ?? 'en',
+  );
+
+  useEffect(() => {
+    const unsubscribe = onLanguageChange((newLang: string) => {
+      setLanguageState(currentLang =>
+        currentLang !== newLang ? newLang : currentLang,
+      );
+    });
+
+    return () => {
+      if (unsubscribe) {
+        unsubscribe();
+      }
+    };
+  }, []);
+
+  const setLanguage = useCallback(
+    async (newLang: string) => {
+      if (newLang !== languageState) {
+        // Set locale in gmp
+        changeGmpLocale(newLang);
+        gmp.setLocale(newLang);
```

Comment:
I would love to get rid of this unnecessary extra API which internally calls changeGmpLocal (again) and sets the locale in the browser session via the GmpSettings. Therefore we should just use the following here
```suggestion
        // store language in the session
        gmp.settings.locale = newLang;
```